backend/metrics_backend/model/payment_network_metrics.py

```diff
@@ -53,16 +54,8 @@ def handle_token_network_created(self):
         self.num_token_networks += 1
 
     def _add_opened_channel_to_participant(self, participant: Address):
-        if not participant in self.open_channels_by_participant:
-            self.open_channels_by_participant[participant] = 0
         self.open_channels_by_participant[participant] += 1
     
     def _remove_opened_channel_from_participant(self, participant: Address):
-        try:
+        if not self.open_channels_by_participant[participant] == 0:
```

Comment:
I think the check here should be
```py
if not participant in self.open_channels_by_participant)...
```
Otherwise the `KeyError` might still happen.

Reply:
I changed to the defaultdict, so there are no KeyErrors anymore. I am checking for the default value.